falcon/hooks.py

```diff
@@ -79,6 +91,8 @@ def do_something(req, resp, resource, params):
             *action*.
     """
 
+    is_async = kwargs.get('is_async', False)
```

Comment:
if we keep `is_async: bool = False,` listed explicitly this line is not needed

Reply:
I would like to deprecate `is_async` now that Cython 3.0 properly marks coroutine functions if run on Python 3.10+. I will create a separate issue for this.

Reply:
If we can drop `is_async` this PR becomes possible. After I pushed this branch I continued my testing and realized this wont work the way I thought it would otherwise I'm not seeing a way to actually type `before` and `after` methods. I could be missing something though 🤷🏼‍♂️